packages/web/src/components/buy-audio-modal/components/StripeBuyAudioButton.tsx

```diff
@@ -45,57 +33,29 @@ export const StripeBuyAudioButton = () => {
   const belowThreshold =
     purchaseInfo?.isError === false && purchaseInfo.estimatedUSD.uiAmount < 1
 
-  const handleSessionUpdate = useCallback(
-    (e: any) => {
-      if (e?.payload?.session?.status) {
-        dispatch(
-          stripeSessionStatusChanged({ status: e.payload.session.status })
-        )
-        if (e.payload.session.status === 'fulfillment_complete') {
-          dispatch(onRampSucceeded())
-          setIsStripeModalVisible(false)
-        }
-      }
-    },
-    [dispatch, setIsStripeModalVisible]
-  )
-
   const handleClick = useCallback(async () => {
     if (!amount || !purchaseInfo || purchaseInfo?.isError === true) {
       return
     }
     dispatch(onRampOpened(purchaseInfo))
     try {
-      const res = await createStripeSession({
-        amount,
-        destinationWallet: (await getRootSolanaAccount()).publicKey.toString()
-      })
-      if (!STRIPE_PUBLISHABLE_KEY) {
-        throw new Error('Stripe publishable key not found')
-      }
-      const stripeOnRampInstance = await loadStripeOnramp(
-        STRIPE_PUBLISHABLE_KEY
+      const destinationWallet: string = (
+        await getRootSolanaAccount()
+      ).publicKey.toString()
+      dispatch(
+        initializeStripeModal({
+          amount,
+          onRampSucceeded,
+          onRampCanceled,
+          destinationCurrency: 'sol',
+          destinationWallet
+        })
       )
-      if (!stripeOnRampInstance) {
-        throw new Error('Stripe onramp instance not found')
-      }
-      const session = stripeOnRampInstance.createSession({
-        clientSecret: res.client_secret
-      })
-      session.mount('#stripe-onramp-modal')
-      session.addEventListener('onramp_session_updated', handleSessionUpdate)
-      setIsStripeModalVisible(true)
     } catch (e) {
```

Comment:
nit, this try catch feels unnecessary now?

Reply:
Oh actually looking at this again, there is some async code for fetching the root account that could technically fail?
Going to leave this catch for that case.